src/cli/commands/install.js

```diff
@@ -947,6 +947,17 @@ export async function run(config: Config, reporter: Reporter, flags: Object, arg
   if (flags.lockfile === false) {
     lockfile = new Lockfile();
   } else {
+    const haveLockfile = await fs.exists(path.join(config.lockfileFolder, constants.LOCKFILE_FILENAME));
+    if (haveLockfile) {
+      //no lockfile found, test if node_module exists
+      const haveNodeModules = await fs.exists(path.join(config.lockfileFolder, constants.NODE_MODULES_FOLDER));
+      if (haveNodeModules) {
+        if (!config.nonInteractive && !await reporter.questionAffirm(reporter.lang('lockfileModulesConflict'))) {
+          reporter.footer(false);
+          throw new MessageError(reporter.lang('operationAborted'));
+        }
+      }
+    }
     lockfile = await Lockfile.fromDirectory(config.lockfileFolder, reporter);
```

Comment:
`Lockfile.fromDirectory` already checks for the existence of the lockfile and also prints a warning to the reporter if it doesn't exist. It can be modified to branch into a callback when the lockfile does not exist. Alternatively, one can look at `lockfile.parseResult` to see if the file was actually parsed or not (empty `parseResult` field when the file is missing).

This would make the code a bit more clear and would also reduce one extra I/O per install for the same file.